aregion.h

```diff
@@ -353,6 +354,9 @@ class ARegion : public AListElem
 		int distance;
 		ARegion *next;
 
+		// A link to the region's level to make some things easier.
+		ARegionArray *level;
+
```

Comment:
this was added here so that a region knows what level it belongs to.  This means that I don't need to pass around the entire pRegionsList everywhere, which (while it hasn't be eradicated from most places yet) will enable that to be removed from things that already know what region they are working on AND will make the code cleaner in the long run.

It was needed here since when storing a region as part of an event, I would otherwise have no way to convert that event into json without *also* storing the regions array pointer in the event, which is just bad coding style :)  And, imho, a region should know which level it belongs to, rather than just knowing the zloc and relying on passing around the entire list of arrays in order to get back to the right right.